### PR TITLE
Add reset view function and button

### LINKED ISSUE #1553: Visualizer: Add resetView function and button
### Description of the task

When the SimualtorGUI or FullSystemGUI start up, we make sure to focus the visualization contents on the field. This can be thought of as the "default view".

As users zoom and pan around the scene, they may want a quick way to reset the view back to the entire field. This would be the purpose of a `resetView` button.

This can likely be implemented in the `DrawFunctionVisualizer`, since it's used for both the FullSystemGUI or SimulatorGUI. This could either be implemented with a button, or a context menu item (the menu that appears during a right-click).

### Acceptance criteria

- [ ] Add a `resetView` button to the DrawFunctionVisualizer

### PR DESCRIPTION
**Pull request by BrianS99**
### Description
Added a Reset View button that can be accessed using context menu on simulator field (right-click)

### Testing Done
Since applies to visual simulator only, tested by running the standalone simulator as well full simulator

### Resolved Issues
Resolves #1553

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.cpp [hunk 1]
```diff
@@ -71,5 +78,31 @@ void DrawFunctionVisualizer::clearAndDraw(const std::vector<DrawFunction> &draw_
 void DrawFunctionVisualizer::setViewArea(const Rectangle &view_area)
 {
     // Moves and scales the view to fit the view_area in the scene
+    lastViewArea = view_area;
+
+    if (remaining_attempts_to_set_view_area > 0)
+    {
```
Comment (akhilveeraghanta) [inline src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.cpp]: why doesn't `fitInView` work the first time?

File: src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp [hunk 2]
```diff
@@ -89,6 +89,8 @@ void StandaloneSimulatorDrawFunctionVisualizer::contextMenuEvent(QContextMenuEve
     auto robot_under_cursor = standalone_simulator->getRobotAtPosition(point_in_scene);
 
     QMenu menu(this);
+    auto resetViewAction = createResetView();
+    menu.addAction("Reset View", resetViewAction);
```
Comment (akhilveeraghanta) [inline src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp]: can we just use `resetView` here? do we need the create function?

File: src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.h [hunk 3]
```diff
@@ -39,4 +39,32 @@ class DrawFunctionVisualizer : public ZoomableQGraphicsView
     // it, so we don't have to
     QGraphicsScene* graphics_scene;
     QOpenGLWidget* open_gl_widget;
+
+   protected:
+    // Functions to be inherited by standalone_simulator_draw_function_visualizer
+
```
Comment (akhilveeraghanta) [inline src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.h]: nit: you can remove this comment, we usually don't talk about subclasses in the parent class

File: src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.h [hunk 4]
```diff
@@ -39,4 +39,32 @@ class DrawFunctionVisualizer : public ZoomableQGraphicsView
...
+    /**
+     * Called to get the reset view function reference
+     *
+     * @return a reference of a function to reset view of the simulator
+     */
+    std::function<void()> createResetView();
+
+    /**
+     * Resets the view of the simulator, using the last saved {@code Rectangle
+     * lastViewArea} as reference
```
Comment (akhilveeraghanta) [inline src/software/gui/generic_widgets/draw_function_visualizer/draw_function_visualizer.h]: nit: java doc needs a @param

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
it works really well! super useful feature :fire: nice work

just a couple comments from my end

**Review by jonathanlew — COMMENTED** [review]
Nothing to add beyond what's already been said

**Comment by BrianS99** [comment]
Hi @akhilveeraghanta, I put the function in the context menu function, and I cleaned up the nits. We originally thought that we needed to call `fitInView` many times, but turns out we just need to call it once to work here

Tested again with the simulators, works well

**Review by itsarune — DISMISSED** [review]
👍👍

**Review by jonathanlew — CHANGES_REQUESTED** [review]
just one nit

**Comment by akhilveeraghanta** [comment]
nothing to add beyond existing comments

**Review by jonathanlew — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
lgtm :fire:

**Pull request merged**